types/hasura/index.d.ts

```diff
@@ -0,0 +1,266 @@
+// Type definitions for Hasura GraphQL Engine API types 0.1
+// Project: https://github.com/hasura/graphql-engine
+// Definitions by: Toan Nguyen <https://github.com/hgiasac>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export const XHasuraAdminSecret = 'x-hasura-admin-secret';
+export const XHasuraClientName = 'hasura-client-name';
+export const XHasuraRole = 'x-hasura-role';
+export const XHasuraUserID = 'x-hasura-user-id';
+export const HASURA_ROLE_ADMIN = 'admin';
+
+export interface ColumnBoolExp<T = string | number | boolean> {
+    _eq?: T;
+    _neq?: T;
+    _gt?: T;
+    _gte?: T;
+    _lt?: T;
+    _lte?: T;
+    _is_null?: boolean;
+    _in?: T[];
+    _nin?: T[];
+}
+
+export interface TextColumnBoolExp extends ColumnBoolExp<string> {
+    _like?: string;
+    _ilike?: string;
+    _nlike?: string;
+    _nilike?: string;
+    _similar?: string;
+    _nsimilar?: string;
+}
+export type JSONColumnBoolExp = ColumnBoolExp<string>;
+export interface JSONBColumnBoolExp extends ColumnBoolExp<string> {
+    _contains?: string;
+    _contained_in?: string;
+    _has_key?: string;
+    _has_keys_all?: string;
+    _has_keys_any?: string;
+}
+
+type Equals<X, Y> = (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y ? 1 : 2 ? true : false;
+type ScalarType = string | number | boolean | ScalarJSON<unknown> | ScalarJSONB<unknown>;
+type ObjectType = Record<string, ScalarType | Record<string, ScalarType> | Array<Record<string, ScalarType>>>;
+export type ScalarJSON<T> = T & {
+    __type?: 'json';
```

Comment:
This looks like it was intended to be a brand and brands aren't usually optional.  Am I missing something?

Reply:
@amcasey My idea is defining a wrapped type from original type `<T>`, so the Typescript's condition check can know it is JSON type. 

I tried to define 
```
type ScalarJSON<T> = T;
```
However the Equals check doesn't work. It still think 2 types are the same
```ts
Equals<T, ScalarJSON<T>> // true
```
So I decide adding an optional field to make them different. The user doesn't need to know or input this field.

However, if you have better idea, I'm really appreciated to know